crypto/keys/bls12381/multisig.go

```diff
@@ -0,0 +1,88 @@
+package bls12381
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	blst "github.com/supranational/blst/bindings/go"
+)
+
+func aggregatePublicKey(pks []*PubKey) *blst.P1Affine {
+	pubkeys := make([]*blst.P1Affine, len(pks))
+	for i, pk := range pks {
+		pubkeys[i] = new(blst.P1Affine).Deserialize(pk.Key)
+		if pubkeys[i] == nil {
+			panic("Failed to deserialize public key")
+		}
+	}
+
+	aggregator := new(blst.P1Aggregate)
+	b := aggregator.Aggregate(pubkeys, false)
+	if !b {
+		panic("Failed to aggregate public keys")
+	}
+	apk := aggregator.ToAffine()
+
+	return apk
+}
+
+// AggregateSignature combines a set of verified signatures into a single bls signature
+func AggregateSignature(sigs [][]byte) []byte {
+	sigmas := make([]*blst.P2Affine, len(sigs))
+	for i, sig := range sigs {
+		sigmas[i] = new(blst.P2Affine).Uncompress(sig)
+		if sigmas[i] == nil {
+			panic("Failed to deserialize signature")
+		}
+	}
+
+	aggregator := new(blst.P2Aggregate)
+	b := aggregator.Aggregate(sigmas, false)
+	if !b {
+		panic("Failed to aggregate signatures")
+	}
+	aggSigBytes := aggregator.ToAffine().Compress()
+	return aggSigBytes
+}
+
+// VerifyMultiSignature assumes public key is already validated
+func VerifyMultiSignature(msg []byte, sig []byte, pks []*PubKey) bool {
```

Comment:
ok, thanks, will change. I was just following what they did for secp256k1 and ed25519.

Reply:
Yep I've already seen sdk dev loves to panic all around the place. Without any comments on those like `// panic here is fine because XYZ` it's hard to tell how safe it is. Up to you to change it, I don't have the whole context in mind!

Reply:
I've made changes according to your comments and recommitted. Could you please have a look? Thanks